### ISSUE TITLE
SQLite - composite foreign keys are not present in table definition

### ISSUE THREAD

**Issue by randsu** [fluentmigrator/fluentmigrator#776]
The fluent syntax:

```
 Create.Table("NewTable")
	.WithColumn("ForeignColumnA").AsInt32().PrimaryKey().ForeignKey("FK_TableA_NewTable", "TableA", "ForeignColumnA")
	.WithColumn("ForeignColumnB").AsInt32().PrimaryKey().ForeignKey("FK_TableB_NewTable", "TableB", "ForeignColumnB");
```

yields (the not functioning):

```
CREATE TABLE NewTable (
	ForeignColumnA	  INTEGER NOT NULL,
	ForeignColumnB	  INTEGER NOT NULL,
	PRIMARY KEY(ForeignColumnA, ForeignColumnB)
);
```

but should instead yield (the functioning):

```
CREATE TABLE NewTable(
	ForeignColumnA    INTEGER NOT NULL,
	ForeignColumnB 		INTEGER NOT NULL,
	PRIMARY KEY (ForeignColumnA, ForeignColumnB)
	FOREIGN KEY(ForeignColumnA)  REFERENCES TableA(ForeignColumnA),
	FOREIGN KEY(ForeignColumnB)  REFERENCES TableB(ForeignColumnB)
)
```

Hence,  the foreign key constraints are not beeing enforced.

**Comment by escarls** [comment]
It doesn't look like the code supports building the SQL for foreign keys when you use SQLite.

**Comment by escarls** [comment]
Also it looks like SQLite is a bit limited in the ALTER TABLE syntax, so the only way to add FK constraints is via the CREATE TABLE syntax.
I haven't looked that much at the code yet, but is seems that FluentMigrator is built to use ALTER TABLE to add FK constraints.
I just realized that you'll get the same issue if you have a multi-column primary key, since FluentMigrator uses ALTER TABLE.

**Comment by randsu** [comment]
Thanks,

ALTER TABLE gives most flexibility, so it's understandable that FluentMigrator uses this approach, also  considering the AutoReversingMigration-possibility, which I think is quite useful.

But it's **dangerous** that no exception is being thrown, you are lead to believe that constraints are being set properly.

I suggest an addition/rewrite in this case, hopefully FluentMigrator can use sql CREATE TABLE statement for the generation of foreign keys for SQLite, in this case, when using **fluent syntax** beginning with Create.Table-expression.

But then again, if we must use temp tables to accomplish a reversal (remove foreign keys), to get around SQLite's limitations, that might be a can of worms that I don't want to open.

**Comment by escarls** [comment]
It seems this have been asked earlier also in #356 and there is a PR to add support for it in #638.
Maybe you could give it a try?

**Comment by randsu** [comment]
Yeah, wonder where that PR ended up... Not in a release, I suppose..

**Comment by escarls** [comment]
It's not merged in, it looks like there was a question about if it would break any existing migrations.

**Comment by fubar-coder** [comment]
Fixed in release/2.0.0
